---
title: "R Notebook_Management of genome-scale data"
output: html_notebook
---

## Loading the GSE5859Subset package and data

```{r}
library(devtools)
install_github("genomicsclass/GSE5859Subset")
library(GSE5859Subset)
data(GSE5859Subset)

```
## Checking the dimensions of the gene expression matrix and data frames

```{r}
dim(geneExpression)
dim(geneAnnotation)
dim(sampleInfo)

```

## Understanding the relationship between entities

# Checking if the filenames in sampleInfo match the column names of geneExpression

```{r}
all.equal(sampleInfo$filename, colnames(geneExpression))
```

# Checking if the row names of geneExpression match the PROBEID field of geneAnnotation
```{r}
all.equal(rownames(geneExpression), geneAnnotation$PROBEID)
```

```{r}
## Setting options for displaying numbers with two digits
## Displaying a subset of coordinated information
# Creating a combined table with sample information and gene expression data

options(digits=2)
cbind(sampleInfo[1:3,], colnames(geneExpression)[1:3], 
    t(geneExpression)[1:3,1:4])
```
```{r}
# Binding the tables together in an ExpressionSet

# Improving annotation for individual components

# Setting row names for sampleInfo and geneAnnotation to improve visibility
rownames(sampleInfo) = sampleInfo$filename
rownames(geneAnnotation) = geneAnnotation$PROBEID
```

```{r}
# Creating the ExpressionSet

# Loading the Biobase library
library(Biobase)
# Creating the ExpressionSet object (es5859) with geneExpression as assayData,
# sampleInfo as phenoData, and geneAnnotation as featureData

es5859 = ExpressionSet(assayData = geneExpression)
pData(es5859) = sampleInfo
fData(es5859) = geneAnnotation
es5859

```

```{r}
# Selecting features using higher-level concepts

# Selecting expression data for genes on the Y chromosome
es5859[which(fData(es5859)$CHR=="Y"),]
```
```{r}
# Exploring methods and important functions

# Listing the most important methods for ExpressionSet instances
methods(class="ExpressionSet")
```
# exprs(): get the numerical expression values
# pData(): get the sample-level data
# fData(): get feature-level data
# annotation(): get a tag that identifies nomenclature for feature names
# experimentData(): get a MIAME-compliant metadata structure









































